lib/supabase/middleware/middleware.test.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @jest-environment node
+ */
+
+import { NextRequest } from 'next/server';
+import { updateSession } from './middleware';
+import { createClient } from '@supabase/supabase-js';
+
+const supabase = createClient(
+  process.env.NEXT_PUBLIC_SUPABASE_URL!,
+  process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
+);
+
+describe('Middleware', () => {
+  it('redirects unauthenticated user back to /', async () => {
+    const request = new NextRequest('https://localhost:4000/dashboard');
```

Comment:
We're testing local or staging?

Reply:
also make the base URL an environment variable and then you can do something like this.


```suggestion
    const request = new NextRequest(`${new URL('/dashboard', BASE_APP_URL)}`);
```

Reply:
I can't make this change yet as the staging/production url has not been set up yet on vercel